core/trino-spi/src/main/java/io/trino/spi/metrics/LongSumValue.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.metrics;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LongSumValue
+        implements Metric
+{
+    private final Long value;
+
+    @JsonCreator
+    public LongSumValue(long value)
+    {
+        this.value = value;
+    }
+
+    @JsonProperty("value")
+    public long getValue()
+    {
+        return value;
+    }
+
+    @Override
+    public Metric merge(Metric other)
```

Comment:
Take a look at `io.trino.util.Mergeable` and implementations of that. We may want to do something similar. Or move Mergeable to the SPI for use in these.

Reply:
Sounds good - I have moved `Mergeable` to `trino-spi` at https://github.com/trinodb/trino/pull/6232/commits/ad3f94ae9739213837a61877d38e430151af18fd and refactored the merging code following the code at `OperatorStats#mergeInfo`.